site/en/docs/extensions/mv3/manifest/web_accessible_resources/index.md

```diff
@@ -6,29 +6,68 @@ updated: 2018-05-14
-interface for `example.com` would allow any resources that interface requires (images, icons,
-stylesheets, scripts, etc.) as follows:
+Web-accessible resources are files inside an extension that can be accessed by web pages or other
+extensions. Extensions typically use this feature to expose images or other assets that need to be
+loaded in web pages, but any asset included in an extension's bundle can be made web accessible.
+
+By default no resources are web accessible; only pages or scripts loaded from an extension's origin
+can access that extension's resources. Extension authors can use the `web_accessible_resources`
+manifest property to declare which resources are exposed and to what origins.
+
+{% Aside %}
+Prior to Manifest V2 all resources within an extension could be accessed from any page on the
+web. This allowed a malicious website to [fingerprint][6] the extensions that a user has installed
+or exploit vulnerabilities (for example [XSS bugs][7]) within installed extensions. 
+
+Beginning with Manifest V2, access to those resources was limited to rotect the privacy of users. MV2
+extensions exposed only those resources explicitly designated as web accessible.
+
+Manifest V3 provides finer-grained control, letting you expose individual resources to specified
+pages, domains, or extensions.
+{% endAside %}
+
+
+## Manifest declaration
+
+An array of objects that declare resource access rules. Each object maps an array of
+extension resources to an array of URLs and/or extension IDs that can access those resources.
 
 ```json
 {
   ...
   "web_accessible_resources": [
-    "images/*.png",
-    "style/double-rainbow.css",
-    "script/double-rainbow.js",
-    "script/main.js",
-    "templates/*"
+    {
+      "resources": [ "test1.png", "test2.png" ],
+      "matches": [ "https://web-accessible-resources-1.glitch.me/*" ]
+    }, {
+      "resources": [ "test3.png", "test4.png" ],
+      "matches": [ "https://web-accessible-resources-2.glitch.me/*" ],
+      "use_dynamic_url": true
+    }
   ],
   ...
 }
 ```
 
-These resources would then be available in a webpage via the URL
-`chrome-extension://[PACKAGE ID]/[PATH]`, which can be generated with the [extension.getURL][1]
-method. Allowlisted resources are served with appropriate [CORS][2] headers, so they're available
+Each object in the array contains these properties:
+<dl>
+  <dt><code>resources</code></dt>
+  <dd>An array of resources to be exposed. Resources are specified as strings. Directories can be exposed using asterisks (e.g. `/images/*` exposes everything in the images directory).</dd>
+  <dt><code>matches</code></dt>
+  <dd>A list of URL match patterns specifying which pages can access the resources. Only the origin is used to match URLs; subdomains patterns (`*.google.com`) and paths are ignored.</dd>
+  <dt><code>extensions</code></dt>
+  <dd>A list of extension IDs, specifying which extensions can access the resources.</dd>
```

Comment:
```suggestion
  <dt><code>resources</code></dt>
  <dd>An array of resources to be exposed. Resources are specified as strings and may contain
  <code>*</code> for wildcard matches. For example, <code>"/images/*"</code> exposes everything in
  the extension's /images directory recursively while <code>"*.png"</code> exposes all PNG
  files.</dd>
  <dt><code>matches</code></dt>
  <dd>A list of URL match patterns specifying which pages can access the resources. Only the origin
  is used to match URLs; subdomains patterns (<code>*.google.com<code>) and paths are ignored.</dd>
  <dt><code>extensions</code></dt>
  <dd>A list of extension IDs, specifying which extensions can access the resources.</dd>
```

Replaces backticks with `<code>` tags, wraps at 100 characters, and clarifies wildcard behavior.